shared/tests/utility/support/TestYamlConversion.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+
+
+#include <Eigen/Core>
+#include <catch.hpp>
+#include <string>
+#include <utility>
+#include <yaml-cpp/yaml.h>
+
+#include "utility/support/yaml_expression.hpp"
+
+
+using utility::support::Expression;
+using utility::support::resolve_expression;
+
+/**
+ * Unit tests for yaml_expression.cpp
+ *
+ * @author Liam Craft
+ */
+SCENARIO("yaml nodes can be converted to a given container type", "[utility][support][yaml_expression]") {
+    GIVEN("A yaml node containing a list of data") {
+        YAML::Node test_values = YAML::LoadFile("tests/yamlConversionTestVals.yaml");
```

Comment:
Our `yaml` files seem to all be named with `PascalCase`, so I suppose we should stick with that
```suggestion
        YAML::Node test_values = YAML::LoadFile("tests/YamlConversionTestVals.yaml");
```